app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java

```diff
@@ -302,12 +308,42 @@ private void compressImage()  {
             return;
         }
 
+        String title = getString(R.string.compress_image) + " " + preferences.getInt("DefaultCompression", 30) + "%)";
+
         final MaterialDialog dialog = new MaterialDialog.Builder(mActivity)
-                .title(R.string.compress_image)
+                .title(title)
                 .customView(R.layout.compress_image_dialog, true)
                 .positiveText(android.R.string.ok)
                 .negativeText(android.R.string.cancel)
-                .build();
+                .onPositive(new MaterialDialog.SingleButtonCallback() {
+                    @Override
+                    public void onClick(@NonNull MaterialDialog dialog, @NonNull DialogAction which) {
+                        final EditText passwordInput = dialog.getCustomView().findViewById(R.id.quality);
+                        final CheckBox cbSetDefault = dialog.getCustomView().findViewById(R.id.cbSetDefault);
+
+                        int check;
+                        try {
+                            check = Integer.parseInt(String.valueOf(passwordInput.getText()));
+                            if (check > 100 || check < 0) {
+                                Snackbar.make(Objects.requireNonNull(mActivity).findViewById(android.R.id.content),
+                                        R.string.invalid_quality,
+                                        Snackbar.LENGTH_LONG).show();
+                            } else {
+                                mQuality = String.valueOf(check);
+                                if (cbSetDefault.isChecked()) {
+                                    SharedPreferences.Editor editor = preferences.edit();
+                                    editor.putInt("DefaultCompression", check);
```

Comment:
Move the string "DefaultCompression" to this file, and use that here:

https://github.com/Swati4star/Images-to-PDF/blob/master/app/src/main/java/swati4star/createpdf/util/Constants.java